pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -266,10 +268,39 @@ in stdenv.mkDerivation {
 
     # "kernel-install" shouldn't be used on NixOS.
     find $out -name "*kernel-install*" -exec rm {} \;
+
+    # Keep only libudev and libsystemd in the lib output.
+    mkdir -p $out/lib
+    mv $lib/lib/security $lib/lib/libnss* $out/lib/
```

Comment:
I actually wonder why

```
./lib/libnss_myhostname.so.2
./lib/libnss_systemd.so.2
./lib/libnss_mymachines.so.2
./lib/libnss_resolve.so.2
```

are not part of the `$lib` output?  Commit history doesnt really help here. But I see no point in _not_ having them in `$lib` ?

Reply:
Yes, I agree. I've, in the past, often questioned what we moved to lib and what not. Eventually I think that even `lib` might not be granular enough. I could think of separating libraries for `udev`, runtime plugins (like nss) etc in their output so the closures of things that aren't really part of a NixOS system will not be present on systems that just use nixpkgs outside of NixOS.